packages/big-design/src/components/Stepper/Stepper.tsx

```diff
@@ -0,0 +1,104 @@
+import { CheckIcon } from '@bigcommerce/big-design-icons';
+import React, { HTMLAttributes, memo, useEffect, useState } from 'react';
+
+import {
+  StyledDash,
+  StyledLight,
+  StyledLightText,
+  StyledMobileLight,
+  StyledMobileLightText,
+  StyledStep,
+  StyledStepper,
+  StyledText,
+} from './styled';
+
+export interface StepProps {
+  number: number;
+  text?: string;
+  state?: 'pending' | 'current' | 'done';
+  isLast?: boolean;
+}
+
+export interface StepperProps extends HTMLAttributes<HTMLDivElement> {
+  steps: Array<string>;
+  currentStep: number;
+}
+
+export interface LightProps {
+  active?: boolean;
+}
+export interface DashProps {
+  active?: boolean;
+}
+export interface TextProps {
+  active?: boolean;
+}
```

Comment:
If they have the same props, might want to name the interface something a bit more generic to reuse. In my branch I didn't use these, in favor of using dynamic props by extending our utility components, [like so](https://github.com/Spawbreaker/big-design/compare/stepper...chanceaclark:stepper-updates#diff-0e9b4bcc8c6ecb6bd81b845339f2ec126e6b5eb7bcb5f8930fb47d7e5671981dR26).